avocado/utils/linux_modules.py

```diff
@@ -39,17 +39,18 @@
 BUILTIN = 2
 
 
-def load_module(module_name):
+def load_module(module_name, parameters=''):
     """
     Checks if a module has already been loaded.
     :param module_name: Name of module to check
+    :param parameters: Additional parameters to load module with
     :return: True if module is loaded, False otherwise
     :rtype: Bool
     """
     if module_is_loaded(module_name):
         return False
 
-    process.system('/sbin/modprobe ' + module_name)
+    process.system('/sbin/modprobe %s %s' % (module_name, parameters))
```

Comment:
Considering the above difference, I do not see (3) being preferred over (4).
Also, the driver for disk on which /root sits can any way not be unloaded. `rmmod` throws an error saying the module is in use.

If (5) is only for raising exception for generic unload/load, it can be done.